WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -193,26 +198,19 @@ def register_require_sign_in
   end
 
   def register
-    @competition = competition_from_params
-    @registration = nil
-    @selected_events ||= []
-    if current_user
-      @registration = Registration.find_by(user_id: current_user.id, competition_id: @competition.id) || @competition.registrations.build(user_id: current_user.id)
-      @selected_events = @registration.saved_and_unsaved_events.empty? ? @registration.user.preferred_events : @registration.saved_and_unsaved_events
-    end
+    set_registration_info!
   end
```

Comment:
[This](https://github.com/jfly/worldcubeassociation.org/blob/77332b7b53a5a77aa3729bf421f2da9041d0b982/WcaOnRails/app/controllers/registrations_controller.rb#L284) should do the job unless I miss something. With the current code I have a feeling that we lost some information like the events user selected during the update (as we find @registration and override it).